spring-cloud-dataflow-admin-starter/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamDefinitionController.java

```diff
@@ -0,0 +1,251 @@
+
+	private final StreamDeploymentController deploymentController;
+
+	/**
+	 * Create a {@code StreamController} that delegates
+	 * <ul>
+	 *     <li>CRUD operations to the provided {@link StreamDefinitionRepository}</li>
+	 *     <li>module coordinate retrieval to the provided {@link ArtifactRegistry}</li>
+	 *     <li>deployment operations to the provided {@link ModuleDeployer}</li>
+	 * </ul>
+	 *
+	 * @param repository  the repository this controller will use for stream CRUD operations
+	 * @param deploymentController    collabor
+	 * @param deployer    the deployer this controller will use to deploy stream modules
+	 */
+	@Autowired
+	public StreamDefinitionController(StreamDefinitionRepository repository, StreamDeploymentController deploymentController,
+	                                  @Qualifier("processModuleDeployer") ModuleDeployer deployer) {
+		Assert.notNull(repository, "repository must not be null");
+		Assert.notNull(deploymentController, "deploymentController must not be null");
+		Assert.notNull(deployer, "deployer must not be null");
+		this.deploymentController = deploymentController;
+		this.repository = repository;
+		this.deployer = deployer;
+	}
+
+	/**
+	 * Return a page-able list of {@link StreamDefinitionResource} defined streams.
+	 *
+	 * @param pageable   page-able collection of {@code StreamDefinitionResource}.
+	 * @param assembler  assembler for {@link StreamDefinition}
+	 * @return list of stream definitions
+	 */
+	@RequestMapping(value = "", method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
+	public PagedResources<StreamDefinitionResource> list(Pageable pageable,
+			PagedResourcesAssembler<StreamDefinition> assembler) {
+		return assembler.toResource(repository.findAll(pageable), streamAssembler);
+	}
+
+	/**
+	 * Create a new stream.
+	 *
+	 * @param name    stream name
+	 * @param dsl     DSL definition for stream
+	 * @param deploy  if {@code true}, the stream is deployed upon creation
+	 */
+	@RequestMapping(value = "", method = RequestMethod.POST)
+	@ResponseStatus(HttpStatus.CREATED)
+	public void save(@RequestParam("name") String name,
+			@RequestParam("definition") String dsl,
+			@RequestParam(value = "deploy", defaultValue = "true")
+			boolean deploy) throws Exception{
+		if (this.repository.exists(name)) {
+			throw new DuplicateStreamException(
+					String.format("Cannot create stream %s because another one has already " +
+							"been created with the same name", name));
+		}
+
+		StreamDefinition stream = new StreamDefinition(name, dsl);
+		this.repository.save(stream);
+		if (deploy) {
+			deploymentController.deploy(name, null);
+		}
+	}
+
+	/**
+	 * Request removal of an existing stream definition.
+	 *
+	 * @param name the name of an existing stream definition (required)
+	 */
+	@RequestMapping(value = "/{name}", method = RequestMethod.DELETE)
+	@ResponseStatus(HttpStatus.OK)
+	public void delete(@PathVariable("name") String name) throws Exception {
```

Comment:
I believe we don't need to throw exception in most of the REST endpoints in this class and can probably be removed in all the methods where they aren't relevant.